src/grpc/protocol/keyprovider_structs.rs

```diff
@@ -17,34 +30,181 @@ pub struct KeyProviderInput {
     pub keyunwrapparams: KeyUnwrapParams,
 }
 
-#[derive(Serialize, Deserialize, Debug)]
+impl KeyProviderInput {
+    pub fn valid(&self) -> Result<()> {
+        match self.op.as_str() {
+            OP_KEY_WRAP => Err(anyhow!("{}: {:?}", ERR_UNSUPPORTED_OP, self.op)),
```

Comment:
We might want this to be a more verbose message, suggesting that the user needs to use a different keyprovider (such as the sample kbs) to encrypt their images.

Reply:
@fitzthum - Thanks. I've updated the message.